### PR TITLE
[BugFix] fix ngram bloom filter's bug

### PR DESCRIPTION
**Pull request by before-Sunrise**
## Why I'm doing:
ngram bloom filter's logic can affect normal bloom filter index, if only create normal bloom filter index, and query “select xxx from table where cola like xxx”, it will always return empty set. can see sqltest case for details

so in this pr, rename the existed bloom filter index to original_bloom_filter to distinguish from ngram bloom filter. If one function's logic is shared by original_bloom_filter and ngram_bloom_filter, it will be called as "bloom_filter".

For example: BloomFilterIndexWriter has two implementation: 1.OridinaryBloomFilterIndexWriterImpl 2.NgramBloomFilterIndexWriterImpl. They have different logic when writing data into BloomFilter.

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by srlch — APPROVED** [review]

**Review by meegoo — DISMISSED** [review]

File: test/sql/test_index/T/test_ngram_bloom_filter [hunk 1]
```diff
@@ -120,3 +120,26 @@ show index from ngram_index_char;
 
 insert into ngram_index_char values ('2023-01-01',"chinese",3),('2023-01-02',"chineaaa",4),('2023-01-03',"我爱chin",4),('2023-01-04',"toniggrht",4);
 select * from ngram_index_char where username like '_hiaa%';
+
+-- name: test_bloom_filter
+create database ngram_bloom_filter_db_4;
+use ngram_bloom_filter_db_4;
+-- 创建一个包含bloom filter的表
```
Comment (kangkaisen) [inline test/sql/test_index/T/test_ngram_bloom_filter:127]: English

Comment (before-Sunrise) [reply test/sql/test_index/T/test_ngram_bloom_filter:127]: > English

forget to delete this

**Review by satanson — DISMISSED** [review]

File: be/src/storage/rowset/bloom_filter_index_writer.cpp [hunk 2]
```diff
@@ -107,15 +107,15 @@ inline void update_bf(BloomFilter* bf, const typename CppTypeTraits<type>::CppTy
 // Meanswhile, It adds an ordinal index to load bloom filter index according to requirement.
 //
 template <LogicalType field_type>
-class BloomFilterIndexWriterImpl : public BloomFilterIndexWriter {
+class OridinaryBloomFilterIndexWriterImpl : public BloomFilterIndexWriter {
```
Comment (kangkaisen) [inline be/src/storage/rowset/bloom_filter_index_writer.cpp:110]: Original.  should keep consistent.

**Review by kangkaisen — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by before-Sunrise** [comment]
@Mergify backport branch-3.3

**Pull request merged**